src/main_thread/tracks_store/tracks_store.ts

```diff
@@ -481,17 +498,119 @@ export default class TracksStore extends EventEmitter<ITracksStoreEvents> {
       if (
         periodObj.isPeriodAdvertised &&
         trackObj.dispatcher !== null &&
-        !trackObj.dispatcher.hasSetTrack() &&
-        trackObj.storedSettings !== undefined
+        !trackObj.dispatcher.hasSetTrack()
       ) {
-        trackObj.dispatcher.updateTrack(trackObj.storedSettings);
+        trackObj.dispatcher.updateTrack(trackObj.storedSettings ?? null);
       }
       if (this._isDisposed) {
         return;
       }
     }
   }
 
+  /**
+   * Handle the noPlayableRepresentation event, trigger an error if no fallback is possible.
+   * and can trigger event "noPlayableTrack"
+   * @param period - The period that has no playable representation
+   * @param bufferType - The media type that is not playable
+   */
+  private onNoPlayableRepresentation(period: IPeriodMetadata, bufferType: IBufferType) {
+    const periodHasAdaptationForType =
+      period.adaptations[bufferType] !== undefined &&
+      period.adaptations[bufferType].length > 0;
+    const firstPlayableAdaptation = findFirstPlayableAdaptation(period, bufferType);
+
+    if (!periodHasAdaptationForType) {
+      log.debug(
+        `TS: The period does not have adaptation for ${bufferType} there is no track to choose`,
+      );
+      return;
+    }
+
+    if (bufferType === "text") {
+      return;
+    }
+
+    if (
+      firstPlayableAdaptation === undefined &&
+      bufferType === "audio" &&
+      this.onAudioTracksNotPlayable === "continue"
+    ) {
+      // Audio is not playable but video may be playable, let's continue the playback.
+      log.warn(`TS: No playable audio, continuing without audio`);
+      this.trigger("noPlayableTrack", {
```

Comment:
/!\ this event might call the application's code synchronously and completely change the tracksStore's state:
  - it might be disposed
  - the track might have completely changed

I signal this because I've seen no guard against those possibilities, we just assume we're in the same state.